src/buildExec.ts

```diff
@@ -78,18 +84,21 @@ const buildCommitExec = async (): Promise<{
 }> => {
   const commitParent = core.getInput('commit_parent');
   const gitService = getGitService();
-  const overrideBranch = core.getInput('override_branch');
+  let overrideBranch = core.getInput('override_branch');
   const overrideCommit = core.getInput('override_commit');
   const overridePr = core.getInput('override_pr');
   const slug = core.getInput('slug');
   const token = await getToken();
+  if (!overrideBranch && token == null) {
+    overrideBranch = context.payload.pull_request?.head.label;
```

Comment:
i think we should move the "are we a fork" stuff out of `getToken()` and into a new `getOverrideBranch()` function. whether we're a fork doesn't really factor into what the token is, just whether we want to set an override branch. but since the fork check happens inside of `getToken()`, we have to smuggle the result out of `getToken()` by saying `null` means "fork" and `""` means "no fork" so that the override branch code can decide what to do. we should probably move the fork check to where it's really needed instead
```
const getOverrideBranch = async (token: string, context: Whatever): Promise<string | null> => {
  const overrideBranch = core.getInput("override_branch");
  if (!overrideBranch && !token && isPullRequestFromFork()) {
    return Promise.resolve(context.payload.pull_request?.head.label);
  } else {
    return Promise.resolve(overrideBranch);
  }
}

const token = await getToken();
const overrideBranch = getOverrideBranch(token, context);
```

with this change, `getToken()` really just gets the token, and `getOverrideBranch()` contains all of the logic about what the override branch should be, including the fork check